subsys/net/openthread/report/report_generation_utils.cmake

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Usage:
+#   ot_report_git_version(<output> <directory>)
+#
+# This function uses the git describe command to read the current revision
+# of the <directory> directory and saves the result in the <output> variable
+#
+#
+# directory: Directory that contains checked out the git repository of which
+#            you want to read the revision.
+# output: Output variable that will contain the read revision.
+#
+function(ot_report_git_version output directory)
+    execute_process(
+        COMMAND git describe --dirty --always
+        WORKING_DIRECTORY ${directory}
+        OUTPUT_VARIABLE GIT_REV OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_QUIET
+    )
+    set(${output} "${GIT_REV}" PARENT_SCOPE)
+endfunction()
```

Comment:
I'm wondering if over here it wouldn't be better to use only OT commit SHA (maybe even a short one with `git rev-parse --short HEAD`)
- it is used for certification application, so having it in the same format could be helpful 
- OT's tags are managed upstream, so adding them to the report might cause some confusion

@LuDuda what are your thoughts on this?

Reply:
What about having these two values?